### DISCUSSION TITLE
Cannot access a disposed object Session when updating a contentItem

### DISCUSSION THREAD

**Discussion by AmadeuAntunes** [OrchardCMS/OrchardCore#8429] [Q&A]
I'm trying to update a value in a content Item.
```

       await ContentManager.UpdateAsync(myContentItem1);

        await ContentManager.UpdateAsync(myContentItem2);
```
myContentItem1   is not null
myContentItem2   is not null

after call this function and pass my ''myContentItem2' after had called the first time with myContentItem1 I got this error

> {"Cannot access a disposed object.\r\nObject name: 'Session'."}

What is the best way to update a ContentItem in orchard ?

**Comment by deanmarcussen** [comment]
Normally this error would happen if you have not awaited something, or if you have resolved the `ContentManager` in a singleton somewhere, and are then trying to access a disposed session.

The `Session` (underyling YesSql) and `ContentManager` are scoped, and disposed at the end of an http request.

Where are you resolving the `ContentManager` from, i.e. is it a Singleton service somewhere?

**Reply by AmadeuAntunes** [reply]
I'm getting it by dependence injection from class constructor ,  Is there any  way to get  instance from ` IContentManager`

**Reply by AmadeuAntunes** [reply]
I try use` 	var contentManager = ServiceProvider.GetRequiredService<IContentManager>(); `but if I try this more then one time I got some error but now in that line **System.ObjectDisposedException: 'Cannot access a disposed object.**

**Reply by AmadeuAntunes** [reply]
Well I just replace await ContentManager.UpdateAsync(contentItemSms) to ContentManager.UpdateAsync(contentItemSms).Wait(); and seems work
